common/persistence/sql/sql_execution_store.go

```diff
@@ -375,6 +376,24 @@ func (m *sqlExecutionStore) GetWorkflowExecution(
 			Message: fmt.Sprintf("GetWorkflowExecution: failed. Error: %v", err),
 		}
 	}
+	// if we have checksum, we need to make sure the rangeID did not change
+	// if the rangeID changed, it means the shard ownership might have changed
+	// and the workflow might have been updated when we read the data, so the data
+	// we read might not from a consistent view, the checksum validation might fail
+	// in that case, we need to return an error
+	if state.ChecksumData != nil {
+		row, err := m.db.SelectFromShards(ctx, &sqlplugin.ShardsFilter{ShardID: int64(m.shardID)})
+		if err != nil {
+			return nil, convertCommonErrors(m.db, "GetWorkflowExecution", "", err)
+		}
+		if row.RangeID != request.RangeID {
+			return nil, &persistence.ShardOwnershipLostError{
```

Comment:
`TestCreateWorkflowExecutionConcurrentCreate` is timing out. Probably retrying for this case forever.

Reply:
Fixed, it's because the original context was overwritten by a child context.